rmgpy/tools/merge_models.py

```diff
@@ -160,4 +161,17 @@ def combine_models(models):
                   '#{0:d}.'.format(i+1, Nrxn - Nrxn0, len(model.reactions)))
     print('The merged model has {0:d} species and {1:d} reactions'
           ''.format(len(final_model.species), len(final_model.reactions)))
+
+    # ensure no species with same name and index
+    label_index_dict = {}
+    for s in final_model.species:
+        if s.label not in label_index_dict.keys():
+            label_index_dict[s.label] = [s.index]
+        else:
+            if s.index in label_index_dict[s.label]:
+                # obtained a duplicate
+                s.index = max(label_index_dict[s.label]) + 1
```

Comment:
Do we want to have unique indices? In other words, do we want to reindex starting from the largest index in the entire model + 1?

Reply:
This code isn't guaranteeing unique indexes at all. To do that, we would pretty much have to renumber everything except the first model, which would make this PR much less useful.

From my point of view, if the user starts with an un-indexed mechanism, like aramcomech, the output should also not have indexes.

Reply:
Is there a benefit to unique indices that I've overlooked? My main goal for this PR was to not have any CHEMKIN or cantera strings conflict, while keeping those names as similar as possible to the input names.

Reply:
To give an example of the functioning in this PR: if model 1 has `CH3(15)` and model 2 has `H2O(15)`, then neither of their names would be changed, since both the name and index don't match.

I would think for many users, avoiding chemkin or cantera names changing would be preferable to ensuring every species has a unique number.

Reply:
Let me know your thoughts on this

Reply:
Ah, I forgot that we would also need to check for species which already have the same indices.

I think it would only matter if you were merging RMG models and wanted to load the merged model back into RMG, which would parse the indices from the labels. I'm not sure exactly what effect duplicate indices would have in that scenario though. It might not actually matter.

I'm ok with the behavior you implemented. I think the key desired outcome is that the resulting merged mechanism file is directly simulate-able in Chemkin or Cantera.